src/App.jsx

```diff
@@ -14,33 +15,101 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
-export const App = () => (
-  <div className="section content">
-    <div className="buttons">
-      <button type="button" className="button is-info is-light">
-        Sort alphabetically
-      </button>
+export const App = () => {
+  const [goodsOrdered, setGoodsOrdered] = useState(goodsFromServer);
+  const [isReversed, setIsReversed] = useState(false);
+  const [sortType, setSortType] = useState(null);
 
-      <button type="button" className="button is-success is-light">
-        Sort by length
-      </button>
+  const sortAlphabetically = () => {
+    const sorted = [...goodsOrdered].sort((a, b) => a.localeCompare(b));
 
-      <button type="button" className="button is-warning is-light">
-        Reverse
-      </button>
+    if (isReversed) {
+      sorted.reverse();
+    }
 
-      <button type="button" className="button is-danger is-light">
-        Reset
-      </button>
-    </div>
+    setSortType('alphabetically');
+
+    return setGoodsOrdered(sorted);
+  };
+
+  const sortByLength = () => {
+    const sorted = [...goodsOrdered].sort(
```

Comment:
This is causing the failing test, give it a try:
Click the button `Sort by length`, take a look at the goods, and click AGAIN at it, (without changing the sorting, just click on the already sorted length).
Can you spot any difference?

To fix it, try using `goodsFromServer` when sorting, instead of using `goodsOrdered`

Reply:
Vou checar, obrigada

Reply:
@pedro-ruas  Na verdade eu não consigo ver diferença nenhuma clicando de novo no sorting by length... O que deveria acontecer? Pra mim fica igual

Reply:
Ata, descobri um bugzinho no sortby length se o botao do reverse estiver acionado também